### DISCUSSION TITLE
MQTT Home Assistant

### DISCUSSION THREAD

**Discussion by PKlhomesmart** [frickelzeugs/FingerprintDoorbell#100] [Q&A]
Hallo liebe Bastler,

ich bin vor einiger Zeit auf das tolle Projekt gestoßen. Ich habe auch soweit alles am laufen, wie Fingerabdrücke einlesen und erkennen, was sehr gut klappt. 
Nur leider komme ich mit dem Code von MQTT nicht weiter. Habt ihr da eine Vorschlag für mich, wie Home Assistant die Daten ausließt und richtig verarbeiten kann. 
Ich habe zwar was gefunden, das funktioniert aber leider nicht 100%. Ich würde mich über Eure Hilfe freuen, da ich jetzt schon so weit in dem Projekt bin, würde ich das ganz gerne fertigstellen. 
Evtl. kann ja jemand seinen Code aus Yaml veröffentlichen. 

mqtt: 
  sensor:
  #FingerprintDoorbell
    - name: "FingerprintDoorbell matchName"
      state_topic: "FingerprintDoorbell/matchName"
    - state_topic: "FingerprintDoorbell/matchId"
      name: "FingerprintDoorbell matchId"
    - state_topic: "FingerprintDoorbell/ring"
      name: "FingerprintDoorbell ring"
        
#Fingerprint Doorbell
  switch:
    - unique_id: FingerprintDoorbell-IgnoreTouchRing
      command_topic: FingerprintDoorbell/ignoreTouchRing
      payload_on: "off"
      payload_off: "on"
      name: "FingerprintDoorbell IgnoreTouchRing"
      

![Mqtt Explorer](https://github.com/user-attachments/assets/1a36d0e9-b6e9-4448-8133-a5422916322d)
![Mqtt Code](https://github.com/user-attachments/assets/dcc6cd7c-2a10-4a77-b203-17655191dad7)

**Comment by runningsystemchanger** [comment]
> Hallo liebe Bastler,
> 
> ich bin vor einiger Zeit auf das tolle Projekt gestoßen. Ich habe auch soweit alles am laufen, wie Fingerabdrücke einlesen und erkennen, was sehr gut klappt. Nur leider komme ich mit dem Code von MQTT nicht weiter. Habt ihr da eine Vorschlag für mich, wie Home Assistant die Daten ausließt und richtig verarbeiten kann. Ich habe zwar was gefunden, das funktioniert aber leider nicht 100%. Ich würde mich über Eure Hilfe freuen, da ich jetzt schon so weit in dem Projekt bin, würde ich das ganz gerne fertigstellen. Evtl. kann ja jemand seinen Code aus Yaml veröffentlichen.

Da ich ewig rumprobiert habe bis es lief und ich hier im repo nirgends einen entscheidenden Tipp gefunden habe, hinterlasse ich mal kurz einen Hinweis, wie ich es mit HA 2025.11.1 endlich zum Laufen bekommen habe:

Ich habe das Device [über die MQTT Integration als Subentry angelegt](https://www.home-assistant.io/integrations/mqtt/#configuration) und dort für die folgenden Topics jeweils eine Entity angelegt:
fingerprintDoorbell_ring
fingerprintDoorbell_matchId
fingerprintDoorbell_matchName
fingerprintDoorbell_matchConfidence
fingerprintDoorbell_lastLogMessage

Type of Entity: Sensor
State Topic: (siehe oben)
Value Template: {{value}}

<img width="326" height="363" alt="image" src="https://github.com/user-attachments/assets/b27856a7-3bfa-4c8f-830f-8a86a740c384" />

For our international friends:

I added the device [via the MQTT integration as a subentry](https://www.home-assistant.io/integrations/mqtt/#configuration) and added an entity for each of the following topics:
fingerprintDoorbell_ring
fingerprintDoorbell_matchId
fingerprintDoorbell_matchName
fingerprintDoorbell_matchConfidence
fingerprintDoorbell_lastLogMessage

Type of Entity: Sensor
State Topic: (see above)
Value Template: {{value}}
